src/pytest_recording/network.py

```diff
@@ -15,13 +21,20 @@ class Curl(object):
         """
 
         handle = attr.ib(factory=pycurl.Curl)
+        url = None
 
         def __getattribute__(self, item):
+            handle = object.__getattribute__(self, "handle")
             if _disable_pycurl and item == "perform":
+                if _allowed_hosts is not None:
+                    combined = "(" + ")|(".join(_allowed_hosts) + ")"
+                    if re.match(combined, urlparse(self.url).hostname):
```

Comment:
What if the user will call `perform` immediately after instance creation? `self.url` will be `None`. I think the user should see the error from `pycurl` itself, not an error in our wrapper